cmplibrary/src/test/java/com/sourcepoint/gdpr_cmplibrary/GDPRUserConsentTest.java

```diff
@@ -0,0 +1,182 @@
+                "\"IABTCF_PurposeConsents\":\"111111111100000000";
+
+    private GDPRUserConsent userConsent;
+    private JSONObject jsonConsentsMock;
+    private String consentUUIDMock;
+
+    @Before
+    public void setUp() throws JSONException, ConsentLibException {
+        JSONObject consentsResponse = new JSONObject(jsonData);
+        consentUUIDMock = consentsResponse.getString("uuid");
+        jsonConsentsMock = consentsResponse.getJSONObject("userConsent");
+        userConsent = new GDPRUserConsent(jsonConsentsMock ,consentUUIDMock);
+    }
+
+    @Test
+    public void emptyUserConsentConstructor() {
+        GDPRUserConsent emptyUserConsent = new GDPRUserConsent();
+
+        assertEquals(DEFAULT_EMPTY_UUID, emptyUserConsent.uuid);
+        assertEquals(DEFAULT_EMPTY_CONSENT_STRING, emptyUserConsent.consentString);
+        assertTrue(emptyUserConsent.acceptedCategories.isEmpty());
+        assertTrue(emptyUserConsent.acceptedVendors.isEmpty());
+        assertTrue(emptyUserConsent.legIntCategories.isEmpty());
+        assertTrue(emptyUserConsent.specialFeatures.isEmpty());
+        assertTrue(emptyUserConsent.TCData.isEmpty());
+        assertTrue(emptyUserConsent.vendorGrants.isEmpty());
+    }
+
+
+    @Test
+    public void userConsentUUID() {
+        assertEquals(consentUUIDMock, userConsent.uuid);
+    }
+
+    @Test
+    public void userConsentConsentString() throws JSONException {
+        assertEquals(userConsent.consentString, jsonConsentsMock.getString("euconsent"));
+    }
+
+    @Test
+    public void userConsentAcceptedCategories() throws JSONException {
+        ArrayList<String> acceptedCategories = userConsent.json2StrArr(jsonConsentsMock.getJSONArray("acceptedCategories"));
+        assertEquals(acceptedCategories, userConsent.acceptedCategories);
+    }
+
+    @Test
+    public void userConsentAcceptedVendors() throws JSONException {
+        ArrayList<String> acceptedVendors = userConsent.json2StrArr(jsonConsentsMock.getJSONArray("acceptedVendors"));
+        assertEquals(acceptedVendors , userConsent.acceptedVendors);
+    }
+
+    @Test
+    public void userConsentLegInCategories() throws JSONException {
+        ArrayList<String> legIntCategories = userConsent.json2StrArr(jsonConsentsMock.getJSONArray("legIntCategories"));
+        assertEquals(legIntCategories, userConsent.legIntCategories);
+    }
+
+    @Test
+    public void userConsentSpecialFeatures() throws JSONException {
+        ArrayList<String> specialFeatures = userConsent.json2StrArr(jsonConsentsMock.getJSONArray("specialFeatures"));
+        assertEquals(specialFeatures, userConsent.specialFeatures);
+    }
+
+    @Test
+    public void userConsentTCData() throws JSONException, ConsentLibException {
+        HashMap tcData = CustomJsonParser.getHashMap(jsonConsentsMock.getJSONObject("TCData"));
```

Comment:
Same here, you should rely on the java libs instead of our own. Never never trust your code, it's gonna try to deceive you in many ways (at least when it comes to testing 😅 )